src/main/java/org/openstreetmap/atlas/checks/validation/areas/OverlappingAOIPolygonCheck.java

```diff
@@ -0,0 +1,237 @@
+        {
+            if (this.hasMinimumOverlapProportion(aoi.asPolygon(), area.asPolygon()))
+            {
+                if (aoiFiltersTest(object, area))
+                {
+                    flag.addObject(area);
+                    flag.addInstruction(this.getLocalizedInstruction(0, object.getOsmIdentifier(),
+                            area.getOsmIdentifier()));
+                    this.markAsFlagged(area.getIdentifier());
+                    hasOverlap = true;
+                }
+            }
+        }
+
+        if (hasOverlap)
+        {
+            this.markAsFlagged(object.getIdentifier());
+            return Optional.of(flag);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Uses code from findIntersectionType function in BuildingRoadIntersection
+     * 
+     * @param polygon
+     *            {@link Polygon} to check for intersection
+     * @param otherPolygon
+     *            Another {@link Polygon} to check against for intersection
+     * @return true if the polygon to polygon overlap surface area is greater than minimum
+     */
+
+    private boolean hasMinimumOverlapProportion(final Polygon polygon, final Polygon otherPolygon)
+    {
+
+        Clip clip = null;
+        try
+        {
+            clip = polygon.clip(otherPolygon, Clip.ClipType.AND);
+        }
+        catch (final TopologyException e)
+        {
+            System.out
+                    .println(String.format("Error clipping [%s] and [%s].", polygon, otherPolygon));
+        }
+        // Skip if nothing is returned
+        if (clip == null)
+        {
+            return false;
+        }
+        // Sum intersection area
+        long intersectionArea = 0;
+        for (final PolyLine polyline : clip.getClip())
+        {
+            if (polyline != null && polyline instanceof Polygon)
+            {
+                final Polygon clippedPolygon = (Polygon) polyline;
+                intersectionArea += clippedPolygon.surface().asDm7Squared();
+            }
+        }
+        // Avoid division by zero
+        if (intersectionArea == 0)
+        {
+            return false;
+        }
+        // Pick the smaller building's area as baseline
+        final long baselineArea = Math.min(polygon.surface().asDm7Squared(),
+                otherPolygon.surface().asDm7Squared());
+        final double proportion = (double) intersectionArea / baselineArea;
+
+        return proportion >= this.minimumIntersect;
+    }
+
+    /**
+     * Tests the input {@link AtlasObject} against the list of {@link TaggableFilter}s.
+     *
+     * @param object
+     *            {@link AtlasObject} to be tested
+     * @return true if any of the filters return true
+     */
+    private boolean aoiFiltersTest(final AtlasObject object)
+    {
+        for (final TaggableFilter filter : this.aoiFilters)
```

Comment:
Nit: consider stream approach here to simplify: `return this.aoiFilters.stream().anyMatch(filter -> filter.test(object));`